### PR TITLE
Botany Bean Expansion

### PR DESCRIPTION
**Pull request by the-orange-cow**
## About The Pull Request

Adds two new plants to hydroponics:  Green Beans and Jumping Beans.  Green beans contain a small amount of vitamin and multiver, since they're so healthy.  Jumping beans contain ants, seeing as how real life jumping beans contain insect larva.  They also jump!

Adds two new traits to hydroponics:
Prosophobic Inclination.  This trait is found in green beans and prevents plants with high instability from mutating species naturally.  Wild mutation harvests and the floral somatoray can still grant access to mutations, however.
Symbiotic Resilience.  This trait is found in jumping beans and prevents plants with high instability from mutating stats naturally.

Fixes a few very minor spelling errors.

## Why It's Good For The Game

beans_pull

Credit to Meyhazah on discord for the sprite work!

The bean additions give more variety to what botanists can grow, and opens more culinary opportunities in the future for those that want to expand on legume based dishes.  I kind of spitballed stats and reagents that seemed appropriate for the plants in question, and would be glad to adjust them if anyone deems necessary.

The traits were introduced because there are several situations where a botanist may want a plant with higher instability, but NOT want the possibility of a species and stat mutation.  Currently, the only way to guarantee high instability plants don't mutate species is the hydroponics tray autogrow function.  The autogrow function is a heavy consumer of power, and is often not an option, especially earlier in the round.  This trait acts as a secondary method by which a botanist might selectively neuter certain strains.

## Changelog

:cl:
feature: Adds Green Beans and Jumping Beans to hydroponics!  The Jumping Beans really jump!
feature: Adds the Prosophobic Inclination and Symbiotic Resilience plant traits.  No need to worry about unwanted mutations ever again!
spellcheck: Fixed a few insignificant typos.
/:cl:

### CODE REVIEW

File: code/modules/hydroponics/grown/beans.dm [hunk 1]
```diff
@@ -49,3 +49,58 @@
...
+	reagents_add = list(/datum/reagent/consumable/nutriment = 0.05, /datum/reagent/ants = 0.1) //IRL jumping beans contain insect larve, hence the ants
+	rarity = 20
+
+/obj/item/food/grown/jumpingbeans
+	seed = /obj/item/seeds/greenbean/jump
+	name = "jumping bean"
+	desc = "Umm, what's causing it to move like that?"
+	icon_state = "jumpingbean"
+	foodtypes = FRUIT
+	tastes = list("bugs" = 1)
```
Comment (ArcaneMusic) [inline code/modules/hydroponics/grown/beans.dm:106]: You'll need a blank line at the end of the file.

**Review by ArcaneMusic — COMMENTED** [review]
Overall, seems pretty clean and self contained. Nice work!

**Comment by Shadowflame909** [comment]
power gamers eating their beans for vitamin!!!!

**Comment by BebeYoshi** [comment]
It would be interesting if it also made it unable to mutate stats while at high instability, that way it would be perfect to roll random chems/tratis without autogrow/wasting a lot of power but I guess that's probably going to far, looking great

**Comment by CursedBirb** [comment]
I see one problem. Multiver purges chems. Chems made in plants starts with 75% purity. Vitamin if I recall is not a medical chem(I would have to code dive to check but still multiver is not 100% pure so it will be purged anyway). I suggest changing vitamin for two medical chems to prevent them from being purged

**Comment by the-orange-cow** [comment]
> I see one problem. Multiver purges chems. Chems made in plants starts with 75% purity. Vitamin if I recall is not a medical chem(I would have to code dive to check but still multiver is not 100% pure so it will be purged anyway). I suggest changing vitamin for two medical chems to prevent them from being purged

Yeah, I just played around with it and you're correct about it stopping nutrition gain.  However, I worry that loading it up with a lot of medicines might be pushing it when it comes to balance, considering it is easily available at roundstart.

I'm not familiar with cooking, but from what I understand many recipes don't necessarily carry over all the reagents of their ingredients.  I was mainly thinking that the vitamin component could be used as a starter point to give people making recipes a basis for what they want the finished food to contain, as opposed to the actual uncooked bean itself.

I'll put some thought into alternatives for green bean reagents, although if anyone has any specific recommendations I'd certainly take that into account.

**Review by MrMelbert — APPROVED** [review]

**Comment by the-orange-cow** [comment]
Per request, I changed the trait to also prevent stat mutations caused by instability, I can change back if this is an issue.

**Comment by the-orange-cow** [comment]
Upon receiving feedback, I've split the traits into two:  one that stops species mutations(found on green beans) and one that stops stat mutations(found on jumping beans).

**Pull request merged**